libgadget/Makefile

```diff
@@ -139,10 +138,16 @@ libgadget-utils.a: $(GADGET_UTILS_OBJS)
 libgadget.a: $(GADGET_OBJS)
 	$(AR) rv $@ $(GADGET_OBJS)
 
+.objs/petaio.o: petaio.c config.h $(INCL)
```

Comment:
Looks like a workaround. Why?

Reply:
Ah - it is to ensure that petaio.o and only petaio gets rebuilt when config.h changes, because petaio.c is the file that includes config.h . It is just to declare the extra dependency. I will add a comment.

Reply:
But all .o files already depend on $(INCL),  which contains config.h see Makefile.rules.

Reply:
Not any more - I removed config.h from $(INCL) so that one can meaningfully do incremental builds. Otherwise every time you do a commit it will force config.h to be regenerated, which will force a full rebuild.